src/bootchooser.c

```diff
@@ -809,8 +808,9 @@ static gboolean uboot_set_state(RaucSlot *slot, gboolean good, GError **error)
 set_left:
 
 	key = g_strdup_printf("BOOT_%s_LEFT", slot->bootname);
+	val = g_strdup_printf("%d", r_context()->config->uboot_default_attempts);
```

Comment:
Probably it is better here to move the string conversion to `uboot_env_set` and pass integer arguments instead.

Reply:
I'm not sure I'm following. `uboot_set_env()` is used for `BOOT_ORDER` as well, which necessarily is a string. Are you suggesting introducing something like `uboot_set_env_int()` or something?

Reply:
Ouh, didn't notice we use this for order, too. You're right, then it would become more complicated and would possibly require splitting the function up. Thus the solution you implemented is more straight-forward, yes.